waspc/src/Wasp/Generator/NpmDependencies.hs

```diff
@@ -142,9 +141,11 @@ combineNpmDepsForPackage npmDepsForWasp npmDepsForUser =
   if null conflictErrors && null devConflictErrors
     then
       Right $
+        -- todo(filip): check whether dependency updates and npm install work properly
+        -- todo(filip): reconsider whether we want to change the {sever,web-app}/package.json dynamically
         NpmDepsForPackage
-          { dependencies = waspDependencies npmDepsForWasp ++ remainingUserDeps,
-            devDependencies = waspDevDependencies npmDepsForWasp ++ remainingUserDevDeps
+          { dependencies = Map.elems remainingWapsDeps,
+            devDependencies = Map.elems remainingWaspDevDeps
```

Comment:
Let's discuss what we should do here. As a first step, of course. Option 1 is the better long-term solution.  
1. Always specify the same dependencies in `{sever,web-app}/package.json` and alias them. Wasp code will use our version of the dependencies, while the user's code will use their versions.
2. For each dependency, decide which version we allow.
   - If the user doesn't specify that dependency in their `package.json`, Wasp lists it in `{sever,web-app}/package.json`.
   - If the user specifies that dependency with the correct version in their `package.json`, Wasp installs it in the top-level `node_modules` and doesn't list it in `{sever,web-app}/package.json` (this is what this PR implements).
   - If the user specifies that dependency with the wrong version in their `package.json`, Wasp reports a conflict and refuses to build (this is what we do in the old structure).